File: scala-dubbo/scala-dubbo-consumer/src/main/scala/com/yd/scala/dubbo/controller/HelloController.java
package com.yd.scala.dubbo.controller;

import com.yd.scala.dubbo.client.IHelloService;
import com.yd.scala.dubbo.client.ITestService;
import com.yd.scala.dubbo.client.domain.BooleanVO;
import org.apache.dubbo.config.annotation.Reference;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author created by Zeb灬D on 2020-05-30 22:28
 */
@RestController
public class HelloController {

    @Reference
    private IHelloService helloService;

    @Reference
    private ITestService testService;

    @RequestMapping("/sayHello")
    public String getAll(@RequestParam(name = "uid") String uid) {
        return helloService.sayHello(uid);
    }

    @RequestMapping("/test")
    public BooleanVO test(@RequestParam(name = "name") String name,
                          @RequestParam(name = "success") String success) {
        return testService.test(name, Boolean.valueOf(success));
    }
}

File: scala-elasticsearch/src/main/scala/com/yd/scala/elasticsearch/controller/HelloController.scala
package com.yd.scala.elasticsearch.controller

import org.springframework.web.bind.annotation.{RequestMapping, RestController}

/**
  * @author created by Zeb灬D on 2020-06-20 11:10
  */
@RestController
class HelloController {

  @RequestMapping(value = Array("/hello"))
  def sayScalHello() = {
    "Hello scala boot..."
  }
}
